### PR TITLE
move language specific methods to common base class

### PR DESCRIPTION
**Pull request by jmarolf**
follow up to https://github.com/dotnet/roslyn/pull/50318

### CODE REVIEW

**Comment by jmarolf** [comment]
@CyrusNajmabadi ready for review

File: src/Analyzers/Core/Analyzers/SimplifyLinqExpression/AbstractSimplifyLinqExpressionDiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -156,11 +158,24 @@ public void AnalyzeInvocationOperation(OperationAnalysisContext context, INamedT
...
                 => linqMethods.Any(m => m.Equals(invocation.TargetMethod.ReducedFrom ?? invocation.TargetMethod.OriginalDefinition, SymbolEqualityComparer.Default));
 
-            bool IsWhereLinqMethod(IInvocationOperation invocation)
-                => whereMethod.Equals(invocation.TargetMethod.ReducedFrom ?? invocation.TargetMethod.OriginalDefinition, SymbolEqualityComparer.Default);
+            INamedTypeSymbol? TryGetSymbolOfMemberAccess(IInvocationOperation invocation)
+                => (invocation.Syntax is not TInvocationExpressionSyntax invocationNode ||
+                    SyntaxFacts.GetExpressionOfInvocationExpression(invocationNode) is not TMemberAccessExpressionSyntax memberAccess ||
+                    SyntaxFacts.GetExpressionOfMemberAccessExpression(memberAccess) is not SyntaxNode expression)
+                        ? null
+                        : invocation.SemanticModel?.GetTypeInfo(expression).Type as INamedTypeSymbol;
```
Comment (CyrusNajmabadi) [inline src/Analyzers/Core/Analyzers/SimplifyLinqExpression/AbstractSimplifyLinqExpressionDiagnosticAnalyzer.cs:172]: xN'R UNSWEARns.  roo xurw.

Comment (CyrusNajmabadi) [reply src/Analyzers/Core/Analyzers/SimplifyLinqExpression/AbstractSimplifyLinqExpressionDiagnosticAnalyzer.cs:172]: Can't understand.  Too cute.

File: src/Analyzers/VisualBasic/Analyzers/SimplifyLinqExpression/VisualBasicSimplifyLinqExpressionDiagnosticAnalyzer.vb [hunk 2]
```diff
@@ -2,59 +2,22 @@
...
 Namespace Microsoft.CodeAnalysis.VisualBasic.SimplifyLinqExpression
     <DiagnosticAnalyzer(LanguageNames.VisualBasic)>
     Friend Class VisualBasicSimplifyLinqExpressionDiagnosticAnalyzer
-        Inherits AbstractSimplifyLinqExpressionDiagnosticAnalyzer
+        Inherits AbstractSimplifyLinqExpressionDiagnosticAnalyzer(Of InvocationExpressionSyntax, MemberAccessExpressionSyntax)
+
+        Protected Overrides ReadOnly Property SyntaxFacts As ISyntaxFacts = VisualBasicSyntaxFacts.Instance
 
         Protected Overrides Function TryGetNextInvocationInChain(invocation As IInvocationOperation) As IInvocationOperation
             Return TryCast(invocation.Parent, IInvocationOperation)
```
Comment (CyrusNajmabadi) [inline src/Analyzers/VisualBasic/Analyzers/SimplifyLinqExpression/VisualBasicSimplifyLinqExpressionDiagnosticAnalyzer.vb:20]: consider a comment (on both teh c# and VB side).

**Review by CyrusNajmabadi — APPROVED** [review]
LGTM.  just try to avoid cramming so much into an expressoin that it's hard to break down what it's actually doing :)   some of our brains are old and feeble.

**Pull request merged**